Tools/install.sh

```diff
@@ -99,12 +99,34 @@ elif [ -n "$(command -v dnf)" ]; then
     DIST=fedora
     PKG_INSTALL="sudo dnf install -y"
 else
+    _OK=1
     echo "Unsupported distribution"
+    _REQUIRED_TOOLS=(
+            ccache \
+            cmake \
+            curl \
+            git \
+            make \
+            ninja \
+            unzip \
+            g++ \
+            python3 \
+            pip3 \
+	    wget \
+	    )
+    for _TOOL in "${_REQUIRED_TOOLS[@]}"; do
+	if ! [ -x $(command -v "${_TOOL}") ]; then
+	    _OK=0
+	    echo "Install required tool ${_TOOL}"
+	fi
+    done
```

Comment:
This is a good idea. I'm wondering if this could be added as a general step earlier in the script as a `pre-check` for all platforms?

Reply:
AND provide some generic script functions which can be called from the esp32 install.sh with a list.

Reply:
I will look into implementing the `pre-check` function.